libraries/AC_AttitudeControl/AC_AttitudeControl.h

```diff
@@ -489,6 +494,8 @@ class AC_AttitudeControl {
 
     // Intersampling period in seconds
     float               _dt;
+    // scheduler sample time in microseconds (could be different to rate loop)
+    uint64_t            _sample_time_us;
```

Comment:
I don't understand. This is just set and the pulled from here with no changes. The comment say this is because it would be changed in the future in the rate loop. But https://github.com/ArduPilot/ardupilot/pull/28005#discussion_r1749660960 you say that the dt is not associated with the rate loop at all? So why do we need to set this if the rate loop will not be changing it? The log method could get it direct from the scheduler?

Reply:
Yes, I was thinking this would be a better simplification - would avoid changing the API

Reply:
There are two dt's going on - one for the rate control and one for the attitude control. Currently they are the same, but with a separate rate loop they will be different. The atttitude logging should log the attitude dt, the rate logging should log the rate dt (if necessary). There are also two samples / times going on - one for the atttitude controller and one for the rate controller. Also currently the same, but soon to be different. Again the attitude logging should log the attitude one / time. But I do agree that this could simplified by the logging method just directly calling into the scheduler - I will change it to do that.